contracts/test/_fund.js

```diff
@@ -0,0 +1,151 @@
+const {
+  setBalance,
+  setStorageAt,
+} = require("@nomicfoundation/hardhat-network-helpers");
+const ethrs = require("ethers");
+const addresses = require("../utils/addresses");
+const {
+  parseEther,
+  defaultAbiCoder,
+  keccak256,
+  hexlify,
+  zeroPad,
+  solidityKeccak256,
+} = ethrs.utils;
+const { units } = require("../utils/units");
+const erc20Abi = require("./abi/erc20.json");
+
+/**
+ *
+ * Based on https://blog.euler.finance/brute-force-storage-layout-discovery-in-erc20-contracts-with-hardhat-7ff9342143ed
+ * @export
+ * @param {string} tokenAddress
+ * @return {*}  {Promise<number>}
+ */
+const findBalancesSlot = async (tokenAddress) => {
+  const { ethers, network } = (await import("hardhat")).default;
+
+  const encode = (types, values) => defaultAbiCoder.encode(types, values);
+
+  const account = ethrs.constants.AddressZero;
+  const probeA = encode(["uint"], [1]);
+  const probeB = encode(["uint"], [2]);
+
+  const token = await ethers.getContractAt(erc20Abi, tokenAddress);
+
+  for (let i = 0; i < 100; i += 1) {
+    const probedSlot = keccak256(encode(["address", "uint"], [account, i]));
+    if (probedSlot.startsWith("0x0")) continue;
+    // remove padding for JSON RPC
+    // while (probedSlot.startsWith("0x0")) {
+    //     probedSlot = `0x${probedSlot.slice(3)}`
+    // }
+
+    const prev = await network.provider.send("eth_getStorageAt", [
```

Comment:
@nomicfoundation/hardhat-network-helpers has a `getStorageAt` function

Reply:
Thanks, that cleans up the code nicely. Have also added the addresses of the first 4 accounts and contracts to speed up funding a bit more: https://github.com/OriginProtocol/origin-dollar/pull/1872/commits/7396c2b63aa9c85a57de6d6a7ecad26e6900f51f